### PR TITLE
Fix fitsdiff with a directory argument

### LINKED ISSUE #6760: Comparing a FITS file with one in a subdirectory
I knew that when running fitsdiff to compare a file in the default directory with one of the same name in a subdirectory that it was necessary to explicitly give the name of the file in the subdirectory, but today I typed the following by mistake:

    fitsdiff dms_230_491_refpix.fits today_prev/

and fitsdiff reported "No differences found."  However, I knew that the files did differ, so I ran fitsdiff again:

    fitsdiff dms_230_491_refpix.fits today_prev/dms_230_491_refpix.fits

and fitsdiff showed several differences.

It's probably relevant that there was only one file in `today_prev/`.

### PR DESCRIPTION
**Pull request by agowardhan**
os.path.abspath(f) gets the absolute path from the current directory (without checking if file exists) rather than os.path.abspath(path/f) which is needed. 

EDIT: Fix #6760

### CODE REVIEW

**Comment by mirca** [comment]
Hi @agowardhan, thanks for your contribution!

Could you please add a unit test or present a code snippet in which the current behavior fails?

Thanks!

**Comment by agowardhan** [comment]
Here is an example where the current code fails 

```
fitsdiff TEMP4/hst_814w_cal1.fits TEMP/TEMP1/TEMP2/
```

which  returns 

```
 fitsdiff: 3.4
 a: TEMP4/hst_814w_cal1.fits
 b: /Users/socrates/Documents/AstroPy/hst_814w_cal1.fits
 Maximum number of different data values to be reported: 10
 Data comparison level: 0.0

No differences found.

```
i.e. it's looking in a different directory than the one specified. Does this work? Sorry, I am very new to astropy. Let me know if you need anything more.

**Comment by pllim** [comment]
@agowardhan , what we need is a unit test; Something akin to #6621 by @saimn. Also, please insert a change log under bug fix for "2.0.3"; You may look at other entries for formatting examples. Thank you!

File: astropy/io/fits/scripts/fitsdiff.py [hunk 1]
```diff
@@ -225,14 +225,14 @@ def match_files(paths):
 
     for path in paths:
         if glob.has_magic(path):
-            files = [os.path.abspath(f) for f in glob.glob(path)]
+            files = [os.path.abspath(os.path.join(path,f)) for f in glob.glob(path)]
```
Comment (saimn) [inline astropy/io/fits/scripts/fitsdiff.py]: This cannot work if `path` is a glob pattern, and anyway `glob` returns the full path so there is no need to join the path here (a unit test will useful to check this).

**Comment by saimn** [comment]
Any news @agowardhan ?

**Comment by agowardhan** [comment]
Ok, so I wrote the test, and updated the changelog, but I am having trouble merging these changes with the old ones.

**Pull request closed without merging**